pyomo/contrib/viewer/model_browser.py

```diff
@@ -170,14 +170,19 @@ def calculate_children(self):
     def calculate(self):
         """Calculate items, applies to expressions and constraints"""
         if isinstance(self.data, _ExpressionData):
-            self._cache_value = value(self.data)
+            try:
+                self._cache_value = value(self.data, exception=False)
+            except ZeroDivisionError:
+                self._cache_value = "Divide_by_0"
         if isinstance(self.data, _ConstraintData) and self.data.active:
             try:
                 self._cache_value = value(self.data.body, exception=False)
-                self._cache_lb = value(self.data.lower, exception=False)
-                self._cache_ub = value(self.data.upper, exception=False)
-            except:
-                pass
+            except ZeroDivisionError:
+                self._cache_value = "Divide_by_0"
+            # I think these are constants, so don't think I need try/except
+            self._cache_lb = value(self.data.lower, exception=False)
+            self._cache_ub = value(self.data.upper, exception=False)
```

Comment:
These are constants, but that doesn't mean that they will not raise divide-by-zero.  Consider:
```
m = ConcreteModel()
m.x = Var()
m.p = Param(mutable=True, initialize=1)
m.c = Constraint(expr = m.x <= 1/m.p)
m.p = 0
```

Reply:
I see how this could be a problem.  I'll add this check, and try to add a test for it also to be sure it works right.